paasta_tools/deployd/workers.py

```diff
@@ -102,7 +105,7 @@ def run(self) -> None:
                         base=2,
                         max_time=6000,
                     )
-                if bounce_again_in_seconds:
+                if bounce_again_in_seconds and failures < self.max_failures:
                     bounce_by = int(time.time()) + bounce_again_in_seconds
```

Comment:
I think this is where we could use some backoff on attempts

Maybe

```
bounce_by = time.time() + bounce_again_in_seconds * attempts ** 2
```
But eventually we'll be past the magical.. 600 seconds or whatever the drain timeout is?

Idk I think attempts and max_failures are kinda the same, both should be purged from the queue eventually otherwise they will continuosly make deployd spin and waste resources

Reply:
Yeah I think I would rather do it based on failures than attempts.  Did you actually see entries with lots of attempts and not many failures..?

Reply:
Yes. Services in a crash loop don't count as hard failures, they will just get tried to bounce every 60 seconds.